src/charm.py

```diff
@@ -136,6 +137,27 @@ def _configure(self, _) -> None:
             self._write_config_file(content=ue_config)
         self._configure_pebble(restart=service_restart_required)
 
+    def _on_start_simulation_action(self, event: ActionEvent) -> None:
+        """Run network traffic simulation.
+
+        The action tries to ping `8.8.8.8` using the UE interface (oaitun_ue1). Working ping
+        guarantees correctness of the deployment.
+        To avoid deadlocks we're sending only 10 packets.
+        """
+        if not self._container.can_connect():
+            event.fail(message="Container is not ready")
+            return
```

Comment:
Should we also check that the config file is created?

Reply:
I've added a service status check. It should work even better.